lucene/core/src/java/org/apache/lucene/index/IndexWriter.java

```diff
@@ -299,7 +300,70 @@ static int getActualMaxDocs() {
   final FieldNumbers globalFieldNumberMap;
 
   final DocumentsWriter docWriter;
-  private final Queue<Event> eventQueue = new ConcurrentLinkedQueue<>();
+  private final CloseableQueue eventQueue = new CloseableQueue();
```

Comment:
Wouldn't it be nicer to make it just Closeable and pass IndexWriter in the constructor (instead of each method)?